src/content/utils.ts

```diff
@@ -1,55 +1,32 @@
-export class PaymentEndpointError extends Error {}
-
-export function checkWalletAddressUrlFormat(
-  pointerOrUrl: string,
-  urlOnly = false
-) {
-  const httpUrl = urlOnly
-    ? pointerOrUrl
-    : pointerOrUrl.replace(/^\$/, 'https://')
+export class WalletAddressFormatError extends Error {}
 
+export function checkWalletAddressUrlFormat(walletAddressUrl: string): void {
   let url: URL
   try {
-    url = new URL(httpUrl)
-    if (!url.protocol.match(/https?:/)) {
-      const spec = urlOnly ? '' : 'either a payment pointer or '
-      // noinspection ExceptionCaughtLocallyJS
-      throw new PaymentEndpointError(
-        `SPSP endpoint must be specified as ${spec}fully resolved https:// url, ` +
-          `got ${JSON.stringify(pointerOrUrl)} `
+    url = new URL(walletAddressUrl)
+    if (url.protocol !== 'https:') {
+      throw new WalletAddressFormatError(
+        `Wallet address URL must be specified as a fully resolved https:// url, ` +
+          `got ${JSON.stringify(walletAddressUrl)} `
       )
     }
   } catch (e) {
-    if (e instanceof PaymentEndpointError) {
+    if (e instanceof WalletAddressFormatError) {
       throw e
     } else {
-      throw new PaymentEndpointError(
-        `Invalid payment pointer/url: ${JSON.stringify(pointerOrUrl)}`
+      throw new WalletAddressFormatError(
+        `Invalid wallet address URL: ${JSON.stringify(walletAddressUrl)}`
       )
     }
   }
 
-  const isPaymentPointer = pointerOrUrl.startsWith('$')
+  const { hash, search, port, username, password } = url
 
-  if (
-    isPaymentPointer &&
-    (url.hash || url.search || url.port || url.username || url.password)
-  ) {
-    throw new PaymentEndpointError(
-      'Payment pointer must not contain ' +
-        'query/fragment/port/username/password elements: ' +
-        JSON.stringify({
-          hash: url.hash,
-          search: url.search,
-          port: url.port,
-          username: url.username,
-          password: url.password
-        })
+  if (hash || search || port || username || password) {
+    throw new WalletAddressFormatError(
+      `Wallet address URL must not contain query/fragment/port/username/password elements. Received: ${JSON.stringify({ hash, search, port, username, password })}`
```

Comment:
Perhaps we can do a comparison with what we allow, instead of showing the technical hash, search etc parts? 
For example: `if (urlWeAreOkWith !== UrlThatUserGave) throw UserGaveExtraStuff`

Reply:
or, we can just accept and ignore rest of the parts (when saving) if we get the same result anyway?

Reply:
> Perhaps we can do a comparison with what we allow, instead of showing the technical hash, search etc parts?
For example: if (urlWeAreOkWith !== UrlThatUserGave) throw UserGaveExtraStuff

Would attaching a link to the Open Payments documentation work?

> or, we can just accept and ignore rest of the parts (when saving) if we get the same result anyway?

I think we should be strict about the format to not cause misconceptions.

Reply:
We can edit the input to what we want and warn the user ("heads up, you pasted this but wallet address is just the URL that looks like this"); instead of adding friction to this user input.
IMO, we should even let input `$paymentPointer` (as can be copied from rafiki.money at present), to reduce this friction.

I think adding link to documentation is still technical enough for users, as technical as showing the `{query,fragment,port}` etc.

Reply:
Let's do that as a follow-up... reducing friction. For now, let's use an error message that gives most info.

Reply:
Ah! Never mind! I thought it was the input in popup!